util/ranger/detacher.go

```diff
@@ -185,19 +186,71 @@ func getPotentialEqOrInColOffset(sctx sessionctx.Context, expr expression.Expres
 	return -1
 }
 
-// extractIndexPointRangesForCNF extracts a CNF item from the input CNF expressions, such that the CNF item
-// is totally composed of point range filters.
+type cnfItemRangeResult struct {
+	rangeResult *DetachRangeResult
+	offset      int
+	// sameLenPointRanges means that each range is point range and all of them have the same column numbers(i.e., maxColNum = minColNum).
+	sameLenPointRanges bool
+	maxColNum          int
+	minColNum          int
+}
+
+func getCNFItemRangeResult(sctx sessionctx.Context, rangeResult *DetachRangeResult, offset int) *cnfItemRangeResult {
+	sameLenPointRanges := true
+	var maxColNum, minColNum int
+	for i, ran := range rangeResult.Ranges {
+		if !ran.IsPoint(sctx) {
+			sameLenPointRanges = false
+		}
+		if i == 0 {
+			maxColNum = len(ran.LowVal)
+			minColNum = len(ran.LowVal)
+		} else {
+			if len(ran.LowVal) > maxColNum {
+				maxColNum = len(ran.LowVal)
+			}
+			if len(ran.LowVal) < minColNum {
+				minColNum = len(ran.LowVal)
+			}
+		}
```

Comment:
We can initiate `minColNum` with `MaxInt` and then use `Max` and `Min` to make the code simpler.